packages/manager/cypress/e2e/core/account/user-permissions.spec.ts

```diff
@@ -904,4 +905,95 @@ describe('User permission management', () => {
+    const mockUser = accountUserFactory.build({
+      username: randomLabel(),
+      restricted: false,
+    });
+
+    const username = randomLabel();
+    const additionalUser = accountUserFactory.build({
+      username: username,
+      email: `${username}@test.com`,
+      restricted: false,
+    });
+
+    const mockUserGrantsUpdated = grantsFactory.build();
+    const mockUserGrants = {
+      ...mockUserGrantsUpdated,
+      global: undefined,
+    };
+
+    mockGetUsers([mockUser, additionalUser]).as('getUsers');
+    mockGetUser(mockUser);
+    mockGetUserGrants(mockUser.username, mockUserGrants);
+    mockGetUserGrants(additionalUser.username, mockUserGrants);
+    mockDeleteUser(additionalUser.username).as('deleteUser');
+
+    // Navigate to Users & Grants page, find mock user, click its "User Permissions" button.
+    cy.visitWithLogin('/account/users');
+    cy.wait('@getUsers');
+
+    mockGetUsers([mockUser]).as('getUsers');
+
+    // Confirm that the "Users & Grants" page initially lists the main and additional users
+    cy.findByText(mockUser.username).should('be.visible');
+    cy.findByText(additionalUser.username)
+      .should('be.visible')
+      .closest('tr')
+      .within(() => {
+        ui.button
+          .findByTitle('Delete')
+          .should('be.visible')
+          .should('be.enabled')
+          .click();
+      });
+
+    // the "Confirm Deletion" dialog opens
+    ui.dialog.findByTitle('Confirm Deletion').within(() => {
+      ui.button.findByTitle('Cancel').should('be.visible').click();
+    });
+    // click the "Cancel" button will do nothing
+    cy.findByText(mockUser.username).should('be.visible');
+    cy.findByText(additionalUser.username).should('be.visible');
+
+    // clickl the "x" button will dismiss the dialog and do nothing
+    cy.findByText(additionalUser.username)
+      .should('be.visible')
+      .closest('tr')
+      .within(() => {
+        ui.button
+          .findByTitle('Delete')
+          .should('be.visible')
+          .should('be.enabled')
+          .click();
+      });
+    ui.dialog.findByTitle('Confirm Deletion').within(() => {
+      cy.get('[data-testid="CloseIcon"]').should('be.visible').click();
+    });
+    cy.findByText(mockUser.username).should('be.visible');
+    cy.findByText(additionalUser.username).should('be.visible');
+
+    // delete the user
+    cy.findByText(additionalUser.username)
+      .should('be.visible')
+      .closest('tr')
+      .within(() => {
+        ui.button
+          .findByTitle('Delete')
+          .should('be.visible')
+          .should('be.enabled')
+          .click();
+      });
+
+    // the "Confirm Deletion" dialog opens
+    ui.dialog.findByTitle('Confirm Deletion').within(() => {
+      ui.button.findByTitle('Delete').should('be.visible').click();
+    });
+    cy.wait(['@deleteUser', '@getUsers']);
+
+    // the user is deleted
+    cy.findByText(additionalUser.username).should('not.exist');
```

Comment:
```suggestion
    ui.toast.assertMessage(`User ${additionalUser.username} has been deleted successfully.`);
    cy.findByText(additionalUser.username).should('not.exist');
```
This wasn't a ticket requirement, but it would be good to have the toast notification covered too!

Reply:
Agreed!